### ISSUE TITLE
template.content is undefined

### ISSUE THREAD

**Issue by dchambers** [tmpvar/jsdom#1065]
If you create a `<template/>` element than it should have a `content` property containing a `DocumentFragment` instance that contains the templates content. Based on #941 and #944 it sounds like this is supported by jsdom, yet if I write code like this:

```js
var jsdom = require("jsdom").jsdom;
var doc = jsdom("<html><body></body></html>");
var templateElem = doc.createElement('template');
templateElem.id = 'my-id';
templateElem.innerHTML = '<span></span>';
doc.body.appendChild(templateElem);
doc.importNode(doc.getElementById('my-id').content, true);
```

then I just get the following error on the final line of code:

```
/home/dominicc/git-dev/template-spike/node_modules/jsdom/lib/jsdom/level2/core.js:227
      newNode = importedNode.cloneNode(deep, function(a, b) {
                            ^
TypeError: Cannot read property 'cloneNode' of undefined
    at HTMLDocument.core.Document.importNode (/home/dominicc/git-dev/template-spike/node_modules/jsdom/lib/jsdom/level2/core.js:227:29)
    at Object.<anonymous> (/home/dominicc/git-dev/template-spike/src/Templates.js:9:5)
```

I'm using jsdom 3.1.2. Have I done something wrong?

**Comment by domenic** [comment]
Oh, hmm, I guess we might not actually support that yet, like #941 implies. We have parsing support but never actually implemented HTMLTemplateElement. I think.

**Comment by dchambers** [comment]
Fair enough. To get my tests running on IE (which also doesn't support the `content` attribute) I ended up using this [template polyfill](http://jsfiddle.net/brianblakely/h3EmY/). I imagine you'd need something similar within jsdom?

**Comment by domenic** [comment]
Yeah, I guess it should work, but honestly it should be a pretty quick patch to get a working HTMLTemplateElement, one of us can probably knock it out soon.

**Comment by dchambers** [comment]
Excellent news :+1:

**Comment by rickychien** [comment]
+1

**Comment by rickychien** [comment]
@domenic Unfortunately, I ran into same problem here. How can I polyfill with HTMLTemplateElement or is there someone working on this issue?

**Comment by domenic** [comment]
@rickychien take a look at CONTRIBUTING.md to learn how to contribute features like HTMLTemplateElement to jsdom.

**Comment by domenic** [comment]
Fixed by 86ef3543de49deb21d75e07179baf5a45a56c7c3!

**Comment by dchambers** [comment]
Good work, @rickychien :+1:
